### PR TITLE
EHPR-71 Email module

### PR DESCRIPTION
**Pull request by hitankar-freshworks**
Draft PR to get early feedback

- [x] email module.
- [x] way to add different emails with their templates in the future.
- [x] 2 ways to send email CHES or Mailtrap.
- [x] endpoint to test if email service is working. we are protect this or remove it in the near future .
- [x] templates are currently broken. WIP.

### CODE REVIEW

**Comment by arranfw** [comment]
I'm confused about the mailtrap/nodemailer implementation, I thought we were pretty locked into using CHES why are we supporting two methods?

**Comment by fwpushan** [comment]
@hitankar-freshworks @arranfw @dbayly-freshworks 
I don't see any issue of using Mailtrap but my biggest concerns are 
- I always advocate same tech stack and identical environment for all three deployment env (dev/test/prod). So, using different approach and service may create quality and performance issue.
- Security concern, in our STRA, we have to explicitly mention Mailtrap. That, can create some confusion in future.
- Testing, consider functionality is working in dev/test env but not in prod. So, this approach may create confusion and difficulties for testing.
- Effort duplication, if we are using two services for same job, then we have to maintain two code base with conditional branching.
N.B. May be I misunderstood, but my idea was after yesterdays discussion that we are considering Mailtrap as alternative of CHESS as better service.

**Comment by hitankar-freshworks** [comment]
@arranfw @fwpushan removed nodemailer and Mailtrap stuff. Service is fully functional

**Review by fwpushan — APPROVED** [review]
LGTM 👍 . Please check why deployment is failing, possibly no ssm param found.

File: terraform/api.tf [hunk 1]
```diff
@@ -16,14 +16,19 @@ resource "aws_lambda_function" "api" {
...
     variables = {
-      NODE_ENV          = var.target_env
-      AWS_S3_REGION     = var.region
-      RUNTIME_ENV       = "hosted"
-      POSTGRES_HOST     = aws_rds_cluster.pgsql.endpoint
-      POSTGRES_DATABASE = aws_rds_cluster.pgsql.database_name
-      POSTGRES_PASSWORD = data.aws_ssm_parameter.postgres_password.value
-      EXPORT_SECRET     = data.aws_ssm_parameter.export_key.value
-      POSTGRES_USERNAME = var.db_username
+      NODE_ENV           = var.target_env
```
Comment (dbayly-freshworks) [inline terraform/api.tf:19]: Nit, but if we want to trick which environment the application is running in, we should use a different variable. NODE_ENV works is unintuitive ways.

**Review by dbayly-freshworks — APPROVED** [review]
Looks fine to me! One nit

File: Makefile [hunk 2]
```diff
@@ -11,6 +11,8 @@ export PROJECT := $(or $(PROJECT),ehpr)
 export NODE_ENV ?= development
 export ENV_NAME ?= dev
 export POSTGRES_USERNAME = freshworks
+export CHES_CLIENT_ID ?= EHPR_SERVICE_CLIENT
+export MAIL_FROM ?= noreply@gov.bc.ca
```
Comment (arranfw) [inline Makefile:15]: `EHPRDoNotReply@gov.bc.ca` is the `from` documented in the EHPR text document, should we use that here instead?

File: apps/api/src/form/entity/form.entity.ts [hunk 3]
```diff
@@ -5,15 +5,16 @@ import { FormPayloadDTO } from '@ehpr/common';
...
   payload!: FormPayloadDTO;
 
   @Column('varchar', { nullable: true })
   confirmationId!: string;
 
   @Exclude()
-  @Column('timestamp', { nullable: true })
-  notifiedAt?: Date;
+  @Column('varchar', { nullable: true })
+  chesId?: string;
```
Comment (arranfw) [inline apps/api/src/form/entity/form.entity.ts:17]: Determined we don't need a `notifiedAt` column? I don't think it was a requirement right>

File: terraform/api.tf (hunk 1 above)
Comment (hitankar-freshworks) [reply terraform/api.tf:19]: Not sure I fully understand.

File: apps/api/src/form/form.service.ts [hunk 4]
```diff
@@ -1,25 +1,55 @@
...
 
-    return await this.formRepository.save(newForm);
+    const savedForm = await this.formRepository.save(newForm);
+
+    const notifiedForm = await this.sendMail(savedForm);
+
+    return notifiedForm;
+  }
+
+  private async sendMail(form: FormEntity) {
```
Comment (arranfw) [inline apps/api/src/form/form.service.ts:34]: `sendMail` seems too generic and sounds like it should be in the mail module, does this function need to be moved or should it be `sendFormConfirmation[Email]`?

File: apps/api/src/form/entity/form.entity.ts (hunk 3 above)
Comment (arranfw) [reply apps/api/src/form/entity/form.entity.ts:17]: would `chesTransactionId` be better?

File: apps/api/src/mail/mail.module.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import { Module } from '@nestjs/common';
+import { TypeOrmModule } from '@nestjs/typeorm';
+import { FormEntity } from 'src/form/entity/form.entity';
+import { MailService } from './mail.service';
+
+@Module({
+  imports: [TypeOrmModule.forFeature([FormEntity])],
```
Comment (arranfw) [inline apps/api/src/mail/mail.module.ts:7]: Nest noob here, does this mean MailModule uses Typeorm? I don't see it used in the service file

**Comment by hitankar-freshworks** [comment]
sorry @arranfw I had two approvals. already merged my PR. I will take notes of your comments and address them in my next PR.

File: apps/api/src/mail/mail.module.ts (hunk 5 above)
Comment (hitankar-freshworks) [reply apps/api/src/mail/mail.module.ts:7]: Good catch. We are not using the entity in the mail. I'll remove it.

File: apps/api/src/form/form.service.ts (hunk 4 above)
Comment (hitankar-freshworks) [reply apps/api/src/form/form.service.ts:34]: Good idea

File: apps/api/src/form/entity/form.entity.ts (hunk 3 above)
Comment (hitankar-freshworks) [reply apps/api/src/form/entity/form.entity.ts:17]: I figured saving timestamp is pointless as both actions happen at the same time. And yes `chesTransactionId` would be better

File: Makefile (hunk 2 above)
Comment (hitankar-freshworks) [reply Makefile:15]: I think we probably should

**Pull request merged**